src/localmodeling.jl

```diff
@@ -6,47 +6,50 @@ export LocalAverageModel,LocalLinearModel,LocalPolynomialModel
 export TSP
 export MSE1,MSEp
 export estimate_param
-abstract type AbstractLocalModel end
 
+"""
+    AbstractLocalModel
+Supertype of methods for making a prediction `y` for the query `q` given it's nearest
+neighbors `xnn` and their respective images `ynn`. The distances between `q` and the `xnn`
+are given in `dists` to allow for a weighted model.
+
+Concrete subtypes:
+  * `LocalAverageModel(n::Int)`   : Compute a weighted average over the `ynn`.
+  * `LocalLinearModel(n::Int, f::F)` : Compute a weighted linear regression over
+    the given nearest neighbors. `f` is a regularization function to counter unstable
+    models.
+"""
+abstract type AbstractLocalModel end
 
 """
-    LocalAverageModel  <: AbstractLocalModel
-A Model functor that computes an estimate `y_pred` for a query `q`.
-Given the nearest neighbors `xnn` and their images `ynn`,
-it averages over `ynn` weighted by the distances of the `xnn` to `q`.
+    LocalAverageModel <: AbstractLocalModel
+    localmodel = LocalAverageModel(n=2)
+    localmodel(q,xnn,ynn,dists) -> y_pred
```

Comment:
Having an example right underneath the function call signature is not in accordance with the Julia style for documentation strings.

`localmodel = LocalAverageModel(n=2)` does not offer any benefit.

` localmodel(q,xnn,ynn,dists) -> y_pred` the user does not need to know about this. They will never use this low lever signature, they will only call high level functions like "TSP" (which will be renamed".

You should replace all 3 lines with the simple line
`LocalAverageModel (n::Int = 2)`

In addition, since we decided to completely ditch the polynomial model, maybe we don't have to say anything about `AbstractLocalModel`, and in the high level docstring just say that method can be either `LocalAverageModel` or `LocalLinearModel`.